pkg/knode-manager/controllers/secret.go

```diff
@@ -0,0 +1,169 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	kubeinformers "k8s.io/client-go/informers"
+	lister "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	"github.com/kosmos.io/kosmos/pkg/knode-manager/adapters"
+	"github.com/kosmos.io/kosmos/pkg/knode-manager/utils"
+)
+
+// SecretController is a controller sync secret from master cluster to client cluster
+type SecretController struct {
+	handler adapters.SecretHandler
+
+	masterLister lister.SecretLister
+	clientLister lister.SecretLister
+	synced       cache.InformerSynced
+	secretQueue  workqueue.RateLimitingInterface
+}
+
+// NewSecretController returns a new *SecretController
+func NewSecretController(adapter adapters.SecretHandler, mInformer, cInformer kubeinformers.SharedInformerFactory) (*SecretController, error) {
+	rateLimiter := workqueue.NewItemExponentialFailureRateLimiter(time.Second, 30*time.Second)
+	secret := &SecretController{
+		handler:     adapter,
+		secretQueue: workqueue.NewNamedRateLimitingQueue(rateLimiter, "Knode secret-controller"),
+	}
+
+	masterInformer := mInformer.Core().V1().Secrets()
+	clientInformer := cInformer.Core().V1().Secrets()
+
+	if _, err := masterInformer.Informer().AddEventHandlerWithResyncPeriod(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: secret.UpdateSecret,
+		DeleteFunc: secret.DeleteSecret,
+	}, 0); err != nil {
+		return nil, err
+	}
```

Comment:
maybe client event handler needed?

Reply:
I think the secret state of the subset cluster is maintained according to the host cluster

Reply:
sometimes it will not be so reliable, how about to seek leader's opinion？

Reply:
You mean that if the secret of the subset cluster is deleted or updated by mistake, we need to join the queue of the host cluster and perform operations to maintain their consistency？

Reply:
yeah, i just think we should sync data status faithfully. once sub cluster secret was deleted by mistake, the host should know it.

Reply:
I got it, thanks. I really didn't think about it.